lsp/src/text_store.rs

```diff
@@ -1,28 +1,40 @@
 use std::{
     collections::HashMap,
+    ops::{Deref, DerefMut},
     sync::{Arc, Mutex, OnceLock},
 };
 
 use lsp_types::Url;
 
-pub struct TextStore {
-    pub texts: HashMap<String, String>,
+type TxtStore = HashMap<String, String>;
+
+pub struct TextStore(TxtStore);
+
+impl Deref for TextStore {
+    type Target = TxtStore;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for TextStore {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
 }
 
 pub static TEXT_STORE: OnceLock<Arc<Mutex<TextStore>>> = OnceLock::new();
+
 pub fn init_text_store() {
-    _ = TEXT_STORE.set(Arc::new(Mutex::new(TextStore {
-        texts: HashMap::new(),
-    })));
+    _ = TEXT_STORE.set(Arc::new(Mutex::new(TextStore(HashMap::new()))));
 }
 
 pub fn get_text_document(uri: Url) -> Option<String> {
-    return TEXT_STORE
+    TEXT_STORE
```

Comment:
please use explicit return !

Reply:
yeah this isn't going to fly for me fyi

Reply:
Fixed in [ba4e324](https://github.com/ThePrimeagen/htmx-lsp/pull/29/commits/ba4e3243c82cf489f171a37d5e36e1737510e237)